backend/src/main/java/ai/verta/modeldb/experimentRun/FutureExperimentRunServiceImpl.java

```diff
@@ -80,10 +82,25 @@ public void getExperimentRunById(
       GetExperimentRunById request,
       StreamObserver<GetExperimentRunById.Response> responseObserver) {
     try {
-      final var response =
-          futureExperimentRunDAO
-              .findExperimentRuns(
-                  FindExperimentRuns.newBuilder().addExperimentRunIds(request.getId()).build())
+      final var requestValidationFuture =
+          InternalFuture.runAsync(
+              () -> {
+                if (request.getId().isEmpty()) {
+                  String errorMessage =
+                      "ExperimentRun ID not found in GetExperimentRunById request";
+                  throw new InvalidArgumentException(errorMessage);
+                }
+              },
+              executor);
```

Comment:
As per the test cases we should have to validate request parameter first and here validation was missing so added it here